src/common/http/url_decode.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <charconv>
+#include <string_view>
+
+namespace http {
+bool url_decode(std::string_view url, char *decoded_url, size_t decoded_url_len) {
```

Comment:
Looking at this. There doesn't seem much reason why this needs to be an inline function in a header file. Can you put it into a proper .cpp file, please?